### PR TITLE
[PLATFORM-563] feat: Implement new pipeline commands

### PR DESCRIPTION
**Pull request by raulb**
# Description of change

Fixes https://meroxa.atlassian.net/browse/PLATFORM-563.

This pull request contains all commands related to pipelines while also migrating them to use the new builder process adding tests to some commands that didn't have them before.

- [x] create
- [x] list
- [x] remove
- [x] update
 
The idea to ultimately release this will be with the use of `MEROXA_V2` env variable.

example:

- `export MEROXA_V2=only` would make only v2 commands to be available
- `unset MEROXA_V2` would leave the CLI as v1 only.
- `export MEROXA_V2=any_value` would make both v1 and v2 commands coexist. We could decide to remove this restriction once we're feature complete on v2.

# Type of change

- [x]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

# How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging

# Demo

**Before this pull-request**

```
❯ .m help
The Meroxa CLI allows quick and easy access to the Meroxa data platform.

Using the CLI you are able to create and manage sophisticated data pipelines
with only a few simple commands. You can get started by listing the supported
resource types:

meroxa list resource-types

Usage:
  meroxa [command]

Available Commands:
  add         Add a resource to your Meroxa resource catalog
  api         Invoke Meroxa API
  billing     Open your billing page in a web browser
  completion  Generate completion script
  connect     Connect two resources together
  create      Create Meroxa pipeline components
  describe    Describe a component
  help        Help about any command
  list        List components
  login       login or sign up to the Meroxa platform
  logout      logout of the Meroxa platform
  logs        Print logs for a component
  open        Open in a web browser
  remove      Remove a component
  update      Update a component
  version     Display the Meroxa CLI version
  whoami      Display the current logged in user


Flags:
      --config string      config file
      --debug              display any debugging information
  -h, --help               help for meroxa
      --json               output json
      --timeout duration   set the client timeout (default 10s)

Use "meroxa [command] --help" for more information about a command
```

**After this pull-request (with `MEROXA_V2=only`)**

```
❯ .m help pipelines
Manage pipelines on Meroxa

Usage:
  meroxa pipelines [command]

Aliases:
  pipelines, pipeline

Available Commands:
  create      Create a pipeline
  list        List pipelines
  remove      Remove pipeline
  update      Update pipeline name, state or metadata

Flags:
  -h, --help   help for pipelines

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

Use "meroxa pipelines [command] --help" for more information about a command.

❯ .m help pipelines create
Create a pipeline

Usage:
  meroxa pipelines create NAME [flags]

Flags:
  -h, --help              help for create
  -m, --metadata string   pipeline metadata

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help pipelines list
List pipelines

Usage:
  meroxa pipelines list [flags]

Aliases:
  list, ls

Flags:
  -h, --help   help for list

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help pipelines remove
Remove endpoint

Usage:
  meroxa pipelines remove NAME [flags]

Aliases:
  remove, rm, delete

Flags:
  -f, --force   skip confirmation
  -h, --help    help for remove

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)

❯ .m help pipelines update
Update pipeline name, state or metadata

Usage:
  meroxa pipelines update NAME [flags]

Examples:

meroxa pipeline update old-name --name new-name
meroxa pipeline update pipeline-name --state pause
meroxa pipeline update pipeline-name --metadata '{"key":"value"}'

Flags:
  -h, --help              help for update
  -m, --metadata string   new pipeline metadata
      --name string       new pipeline name
      --state string      new pipeline state

Global Flags:
      --config string      config file
      --debug              display any debugging information
      --json               output json
      --timeout duration   set the client timeout (default 10s)
```

# Additional references

[Design document](https://www.notion.so/meroxa/CLI-subject-verb-object-fd84c27e3a2444638172f5f892f1b2ce)

### CODE REVIEW

File: cmd/meroxa/root/connectors/connect_test.go [hunk 1]
```diff
@@ -174,3 +174,16 @@ Source connector %q and destination connector %q successfully created!
 		t.Fatalf("expected \"%v\", got \"%v\"", connectors, gotConnectors)
 	}
 }
+
+func TestConnectExecutionNoFlags(t *testing.T) {
```
Comment (raulb) [inline cmd/meroxa/root/connectors/connect_test.go:178]: This wasn't changed on this PR, but I realized I should have added this test before.

File: cmd/meroxa/root/connectors/create_test.go [hunk 2]
```diff
@@ -172,3 +172,16 @@ Connector %q successfully created!
 		t.Fatalf("expected \"%v\", got \"%v\"", cr, gotConnector)
 	}
 }
+
+func TestCreateConnectorExecutionNoFlags(t *testing.T) {
```
Comment (raulb) [inline cmd/meroxa/root/connectors/create_test.go:176]: Same here. I should have written this test before.

**Review by lovromazgon — APPROVED** [review]
LGTM 👍

**Comment by lovromazgon** [comment]
By the way, I see we have a `describe` command for resources, I'm curious why we don't have one for pipelines?

**Comment by raulb** [comment]
@lovromazgon that's a great question. It sounds like something worth looking into 😀

**Pull request merged**